ledger/src/shredder.rs

```diff
@@ -89,22 +90,52 @@ impl Shredder {
         reed_solomon_cache: &ReedSolomonCache,
         stats: &mut ProcessShredsStats,
     ) -> impl Iterator<Item = Shred> {
-        shred::make_merkle_shreds_from_entries(
-            &PAR_THREAD_POOL,
+        let now = Instant::now();
+        let entries = bincode::serialize(entries).unwrap();
+        stats.serialize_elapsed += now.elapsed().as_micros() as u64;
+        Self::make_shreds_from_data_slice(
+            self,
             keypair,
-            entries,
+            &entries,
+            is_last_in_slot,
+            chained_merkle_root,
+            next_shred_index,
+            next_code_index,
+            reed_solomon_cache,
+            stats,
+        )
+        .unwrap()
```

Comment:
would be nice to actually handle this, but I realize this just matches existing behavior

Reply:
yeah I'll rework shredder once legacy shreds are gone. its API is a dumpsterfire, but one fire at a time.